middleware/logger.go

```diff
@@ -0,0 +1,53 @@
+package middleware
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/gol4ng/httpware/v2"
+	middleware2 "github.com/gol4ng/httpware/v2/middleware"
+	"github.com/gol4ng/logger"
+	"github.com/gol4ng/logger-http"
+)
+
+func Logger(log logger.LoggerInterface, opts ...logger_http.Option) httpware.Middleware {
+	o := logger_http.EvaluateServerOpt(opts...)
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(writer http.ResponseWriter, req *http.Request) {
+			startTime := time.Now()
+			ctx := req.Context()
+
+			currentLogger := logger.FromContext(ctx, log)
+			currentLoggerContext := logger_http.FeedContext(o.LoggerContextProvider(req), ctx, req, startTime).Add("http_kind", "server")
```

Comment:
same as above, there is some risk that, when `log` props is used, we add some context on a global logger